homework4/Task1.py

```diff
@@ -0,0 +1,41 @@
+"""
+Write a function that gets file path as an argument.
+Read the first line of the file.
+If first line is a number return true if number in an interval [1, 3)*
+and false otherwise.
+In case of any error, a ValueError should be thrown.
+Write a test for that function using pytest library.
+You should create files required for the testing inside the test run and remove them after the test run.
+(Opposite to previous homeworks when you reused files created manually before the test.)
+Definition of done:
+ - function is created
+ - function is properly formatted
+ - function has positive and negative tests
+ - tests do a cleanup and remove remove files generated by tests
+You will learn:
+ - how to test Exceptional cases
+ - how to clean up after tests
+ - how to check if file exists**
+ - how to handle*** and raise**** exceptions in test. Use sample from the documentation.
+* https://en.wikipedia.org/wiki/Interval_(mathematics)#Terminology
+** https://docs.python.org/3/library/os.path.html
+*** https://docs.python.org/3/tutorial/errors.html#handling-exceptions
+**** https://docs.python.org/3/tutorial/errors.html#raising-exceptions
+"""
+import os.path
+
+
+def read_magic_number(path: str) -> bool:
+    if not os.path.exists(path):
+        raise ValueError
+    try:
+        with open(path) as f:
+            line = float(f.readline())
+            i = 1.0
+            while i < 3:
+                if round(i, 5) == line:
+                    return True
+                i += 0.00001
+            return False
+    except Exception:
+        raise ValueError
```

Comment:
When you catching a broad exception type (like Exception), it is better to keep underlying exception information. You can do it by checking exception attributes of like this:
```python
try:
    1/0
except Exception as e:
    raise ValueError from e

Traceback (most recent call last):
  File "<input>", line 2, in <module>
ZeroDivisionError: division by zero
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "<input>", line 4, in <module>
ValueError
```
Also it is good to show some message with raised exception for debugging.

Reply:
BTW, it is not the most obvious way to check if the number is between 1 and 3)